source/common/stats/symbol_table_impl.cc

```diff
@@ -0,0 +1,86 @@
+#include "common/stats/symbol_table_impl.h"
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Stats {
+
+// TODO(ambuc): There is a possible performance optimization here for avoiding the encoding of IPs,
+// if they appear in stat names. We don't want to waste time symbolizing an integer as an integer,
+// if we can help it.
+StatNamePtr SymbolTableImpl::encode(const absl::string_view name) {
+  SymbolVec symbol_vec;
+  std::vector<absl::string_view> name_vec = absl::StrSplit(name, '.');
+  symbol_vec.reserve(name_vec.size());
+  std::transform(name_vec.begin(), name_vec.end(), std::back_inserter(symbol_vec),
+                 [this](absl::string_view x) { return toSymbol(x); });
+
+  return std::make_unique<StatNameImpl>(symbol_vec, *this);
+}
+
+std::string SymbolTableImpl::decode(const SymbolVec& symbol_vec) const {
+  std::vector<absl::string_view> name;
+  name.reserve(symbol_vec.size());
+  std::transform(symbol_vec.begin(), symbol_vec.end(), std::back_inserter(name),
+                 [this](Symbol x) { return fromSymbol(x); });
+  return absl::StrJoin(name, ".");
+}
+
+void SymbolTableImpl::free(const SymbolVec& symbol_vec) {
+  for (const Symbol symbol : symbol_vec) {
+    auto decode_search = decode_map_.find(symbol);
+    RELEASE_ASSERT(decode_search != decode_map_.end(), "");
```

Comment:
These should not be release asserts. If these are invariants/preconditions, which I believe they are, the Envoy style is to use ASSERT and documentation. RELEASE_ASSERTs are reserved for places where we are eliding error handling, see https://github.com/envoyproxy/envoy/blob/master/STYLE.md#error-handling for full discussion.

Reply:
OK, changed in https://github.com/envoyproxy/envoy/pull/3927/commits/85c5d8be231fd6616d035e68a8ecd242e751aa30.